### ISSUE TITLE
The wind field can be displayed clamp to ground?

### ISSUE THREAD

**Issue by Amber-372** [RaymanNg/3D-Wind-Field#37]
Thank you for this great project.
And I want to know the wind field can be displayed clamp to ground? like the following picture.

**Comment by RaymanNg** [comment]
I don't think you can force the particles to move on the ground easily, because height is varying.

If you want to do that with this demo, at least you need to provide the WebGL program the height information, and set the particle height according to its location.

P.S. I recalled that Cesium provide an API for rendering something on the ground. If performance is not your concern, you can read the Cesium document and try it out.

**Comment by Amber-372** [comment]
> I don't think you can force the particles to move on the ground easily, because height is varying.
> 
> If you want to do that with this demo, at least you need to provide the WebGL program the height information, and set the particle height according to its location.
> 
> P.S. I recalled that Cesium provide an API for rendering something on the ground. If performance is not your concern, you can read the Cesium document and try it out.

If height of every grid was known, is there any easy way to achieve effect like picture above,but in a GPU render way.
the picture above is cesium polyline, so it's performance bad.  It can't run smoothly.

**Comment by RaymanNg** [comment]
I think you need to deal with below two difficulties:
1. How to get the height data. Cesium probably stores it somewhere, but you may need to dig in its source code to find out, or you will need to provide the height data which matches the terrain yourself.
2. How to use height data in WebGL program. Generally grid data is passed through texture in WebGL program, you can read the texture in the Fragment Shader and set the height for each particle

P.S. Sometimes WebGL program suffers from precision problem, I am not sure whether particles can always be bound to the ground correctly and accurately.

**Comment by Amber-372** [comment]
> I think you need to deal with below two difficulties:
> 
> 1. How to get the height data. Cesium probably stores it somewhere, but you may need to dig in its source code to find out, or you will need to provide the height data which matches the terrain yourself.
> 2. How to use height data in WebGL program. Generally grid data is passed through texture in WebGL program, you can read the texture in the Fragment Shader and set the height for each particle
> 
> P.S. Sometimes WebGL program suffers from precision problem, I am not sure whether particles can always be bound to the ground correctly and accurately.

I have the height data already. so what bothers me is difficulty 2.
What I need to render is just a small bound, but lat/lon interval is high precision. And High precision and small bound data seem rendering strangely. but it's another question.

**Comment by wangshouren7** [comment]
> > I think you need to deal with below two difficulties:
> > 
> > 1. How to get the height data. Cesium probably stores it somewhere, but you may need to dig in its source code to find out, or you will need to provide the height data which matches the terrain yourself.
> > 2. How to use height data in WebGL program. Generally grid data is passed through texture in WebGL program, you can read the texture in the Fragment Shader and set the height for each particle
> > 
> > P.S. Sometimes WebGL program suffers from precision problem, I am not sure whether particles can always be bound to the ground correctly and accurately.
> 
> I have the height data already. so what bothers me is difficulty 2. What I need to render is just a small bound, but lat/lon interval is high precision. And High precision and small bound data seem rendering strangely. but it's another question.

i'm sorry to bother,In the past for so long,but I was wondering if you'd solved the problem. If it has been resolved, can you give me some idea of the solution? Thank you!
